### ISSUE TITLE
Data Explorer: Add Affordance for opening a CSV file in a text editor

### ISSUE THREAD

**Issue by jmcphers** [posit-dev/positron#5206]
If you open a CSV file in Positron, you get a very nice data explorer, which is very helpful if you want to explore the data.

However, if you are trying to use the CSV file for something else -- for example, for editing, or to examine the raw text so you can load/parse it correctly in Python or R -- it's not obvious how to do it. It's _possible_ via a context menu...

![Image](https://github.com/user-attachments/assets/3556d961-f5f3-41ac-8ff6-854417941e60)

... but we could make this a lot easier. I think that, when the Data Explorer is showing a CSV file, we should have a button on its toolbar that takes you right to the CSV file for easy access.

Inspired by: https://github.com/posit-dev/positron/discussions/5198

**Comment by wesm** [comment]
Good idea -- it didn't occur to me when working on the auto-open-in-data-explorer that it would make it hard to do raw edits on a CSV file!

**Comment by jthomasmock** [comment]
This would be a good candidate for the newly added Action Bar and we could target that for some of the 2024.01 work in December.

**Comment by jmcphers** [comment]
Came up today on internal discussion:

> Hello. Clicking a csv file from the Explorer pane and having it open with Data Explorer is great.... most of the time.  Sometimes I just want to edit that file and can't find an escape hatch to view it as it would appear in VS Code. Is there a way in Positron to jump out of read-only Data Explorer in this case?

**Comment by wesm** [comment]
It would be awesome to get the existing ad hoc action bar (with the Clear Filters button) moved over to the new Action Bar and add a button for "Open in Editor"

**Comment by nealrichardson** [comment]
I ran into this too. I saw "Open with..." as an option on right-click, but I assumed that would open the file in a different application outside of Positron, like how it works when you do "Open with..." in macOS Finder. Maybe an explicit "Open in editor" option on the right-click menu would be more clear that that's what you do?

**Comment by ntluong95** [comment]
Could the Data Explorer have the "Edit Mode" for CSV file? Often when you are viewing the data, you also do some sorting, filtering and you spot some errors in your data, then you want to make a direct change on the file. I meant the behavior can be similar as you do in Excel.

**Comment by testlabauto** [comment]
### Verified Fixed
    Positron Version(s) : 2025.03.0-4
    OS Version          : OSX

### Test scenario(s) 

Verified manually and automated cases were added.

Link(s) to TestRail test cases run or created:

**Comment by jmcphers** [comment]
> Could the Data Explorer have the "Edit Mode" for CSV file? Often when you are viewing the data, you also do some sorting, filtering and you spot some errors in your data, then you want to make a direct change on the file. I meant the behavior can be similar as you do in Excel.

Currently the Data Explorer is explicitly a read-only view, but feel free to open this as a feature request.
